webapp/data_structures/RELATIONALOP.js

```diff
@@ -26,86 +25,83 @@ function RELATIONALOP() {
     }
     RELATIONALOP.prototype.get = function RELATIONALOP() {
 
-        var options={
-        rule:["Operator:==(0),~=(1),<(2),<=(3),>(4),>=(5)",this.rule],
-	    zcr:["Use zero crossing (no: 0), (yes: 1)" ,this.zcr],
-	    Datatype:["Datatype (1=double 3=int32 ...)" ,this.Datatype],
+        var options = {
+        rule:["Operator: ==(0), ~=(1), <(2), <=(3), >(4), >=(5)", this.rule],
+	    zcr:["Use zero crossing (no: 0), (yes: 1)" , this.zcr],
+	    Datatype:["Datatype (1=double 3=int32 ...)" , this.Datatype],
         }
         return options
     }
     RELATIONALOP.prototype.set = function RELATIONALOP() {
 
-    this.rule = parseInt((arguments[0]["rule"]))
-	this.zcr = parseInt((arguments[0]["zcr"]))
-	this.Datatype = parseInt((arguments[0]["Datatype"]))
-
-	if(this.zcr< 0 || this.zcr >1)
-	{
-                alert("incorrect operator");
-		RELATIONALOP.get();
-	}
+        this.rule = parseInt((arguments[0]["rule"]));
```

Comment:
```suggestion
        var rule = parseInt(arguments[0]["rule"]);
```
Remove the extra pair of brackets. Use local variable first. Update this. after the throw block.